custom_tracing/src/settings.rs

```diff
@@ -0,0 +1,20 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct Settings {
+    /// File name to be watched by custom tracing
+    #[serde(default = "default_tracing_cfg_file")]
+    pub tracing_cfg_file: String,
+}
+
+pub fn default_tracing_cfg_file() -> String {
+    "tracing.cfg".to_string()
+}
+
+impl Settings {
+    pub fn serde_default() -> Self {
```

Comment:
Because you have a default provider for `tracing_cfg_file`, `Settings` can derive `Default`.

Then where it's used can annotate the field with `#[serde(default)]` and don't need to point to an explicit default function.